Possible NodeJS 22.22.2 regression when installing latest NPM

### Version

v22.22.2

### Platform

```text
Darwin M-FC6WJYY430 25.3.0 Darwin Kernel Version 25.3.0: Wed Jan 28 20:56:35 PST 2026; root:xnu-12377.91.3~2/RELEASE_ARM64_T6030 arm64
```

### Subsystem

_No response_

### What steps will reproduce the bug?

```sh
# use the latest node 22.x
nvm install 22.22.2

# activate the node version
nvm use 22.22.2

# install latest npm
npm install -g npm@latest
```

### How often does it reproduce? Is there a required condition?

Always happens.

### What is the expected behavior? Why is that the expected behavior?

NPM should install successfully.

### What do you see instead?

```
npm error code MODULE_NOT_FOUND
npm error Cannot find module 'promise-retry'
npm error Require stack:
npm error - /opt/hostedtoolcache/node/22.22.2/x64/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/rebuild.js
npm error - /opt/hostedtoolcache/node/22.22.2/x64/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/index.js
npm error - /opt/hostedtoolcache/node/22.22.2/x64/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/index.js
npm error - /opt/hostedtoolcache/node/22.22.2/x64/lib/node_modules/npm/node_modules/libnpmfund/lib/index.js
npm error - /opt/hostedtoolcache/node/22.22.2/x64/lib/node_modules/npm/lib/utils/reify-output.js
npm error - /opt/hostedtoolcache/node/22.22.2/x64/lib/node_modules/npm/lib/utils/reify-finish.js
npm error - /opt/hostedtoolcache/node/22.22.2/x64/lib/node_modules/npm/lib/commands/install.js
npm error - /opt/hostedtoolcache/node/22.22.2/x64/lib/node_modules/npm/lib/npm.js
npm error - /opt/hostedtoolcache/node/22.22.2/x64/lib/node_modules/npm/lib/cli/entry.js
npm error - /opt/hostedtoolcache/node/22.22.2/x64/lib/node_modules/npm/lib/cli.js
npm error - /opt/hostedtoolcache/node/22.22.2/x64/lib/node_modules/npm/bin/npm-cli.js

```

### Additional information

Created a minimal reproducible matrix of Node and NPM versions in https://github.com/altbdoor/node-22-npm-regress/actions/runs/23525843427/job/68478996014 .

For brevity:

| Node | NPM | Success? |
| --- | --- | --- |
| `22.22.1` | `^11.12.0`  | OK  |
| `22.22.1` | `^11.11.0`  | OK  |
| `22.22.1` | `latest`    | OK  |
| `22.22.2` | `^11.12.0`  | NOK |
| `22.22.2` | `^11.11.0`  | NOK |
| `22.22.2` | `latest`    | NOK |

Edit: seems to only happen from a new Node 22.22.2 environment, doing `npm install -g npm@latest`. Installing the versions in increment appears to be OK.

```console
$ npm i -g npm@^10

changed 13 packages in 1s

$ npm i -g npm@~11.10

removed 48 packages, and changed 92 packages in 1s

$ npm i -g npm@~11.11

removed 1 package, and changed 26 packages in 902ms

$ npm i -g npm@~11.12

removed 7 packages, and changed 22 packages in 905ms

$ npm i -g npm@latest

changed 10 packages in 1s
```

we're seeing this too
```
 code MODULE_NOT_FOUND
 Cannot find module 'promise-retry'
 Require stack:
 - .nvm/versions/node/v22.22.2/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/rebuild.js
 - .nvm/versions/node/v22.22.2/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/index.js
 - .nvm/versions/node/v22.22.2/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/index.js
 - .nvm/versions/node/v22.22.2/lib/node_modules/npm/node_modules/libnpmfund/lib/index.js
 - .nvm/versions/node/v22.22.2/lib/node_modules/npm/lib/utils/reify-output.js
 - .nvm/versions/node/v22.22.2/lib/node_modules/npm/lib/utils/reify-finish.js
 - .nvm/versions/node/v22.22.2/lib/node_modules/npm/lib/commands/install.js
 - .nvm/versions/node/v22.22.2/lib/node_modules/npm/lib/npm.js
 - .nvm/versions/node/v22.22.2/lib/node_modules/npm/lib/cli/entry.js
 - .nvm/versions/node/v22.22.2/lib/node_modules/npm/lib/cli.js
 - .nvm/versions/node/v22.22.2/lib/node_modules/npm/bin/npm-cli.js
 A complete log of this run can be found in: .npm/_logs/2026-04-28T09_23_35_035Z-debug-0.log
```

--- a/.github/actions/install/action.yml
+++ b/.github/actions/install/action.yml
@@ -12,13 +12,13 @@ runs:
     - name: Setup Node.js based on .node-version
       uses: actions/setup-node@53b83947a5a98c8d113130e565377fae1a50d02f # v6.3.0
       with:
-        node-version-file: '.node-version'
-        registry-url: 'https://registry.npmjs.org'
-        cache: 'pnpm'
+        node-version-file: ".node-version"
+        registry-url: "https://registry.npmjs.org"
+        cache: "pnpm"
 
     # Ensure npm 11.5.1 or later is installed
     - name: Update npm
-      run: npm install -g npm@latest
+      run: npm install -g npm@11
       shell: bash
 
     - name: Install Dependencies

